pkg/probe/http/http.go

```diff
@@ -22,20 +22,67 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/url"
+	"path/filepath"
 	"time"
 
 	utilnet "k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/kubernetes/pkg/probe"
 	"k8s.io/kubernetes/pkg/version"
 
+	"crypto/x509"
 	"github.com/golang/glog"
 )
 
-func New() HTTPProber {
+const (
+	defaultKubeletClientCertificateFile = "kubelet-client.crt"
+	defaultKubeletClientKeyFile         = "kubelet-client.key"
+	defaultKubeletCACertificateFile     = "kubelet-ca.crt"
+)
+
+func NewHttpProber() HTTPProber {
 	tlsConfig := &tls.Config{InsecureSkipVerify: true}
 	return NewWithTLSConfig(tlsConfig)
 }
 
+func NewHTTPSProber(certDir string) HTTPProber {
+	if certDir == "" {
+		glog.Errorln("path to HTTPS prober key/cert is not set")
+		return NewHttpProber()
+	}
+
+	certPath, err := filepath.Abs(filepath.Join(certDir, defaultKubeletClientCertificateFile))
+	if err != nil {
+		glog.Errorf("unable to build HTTPS prober cert path: %v", err)
+		return NewHttpProber()
+	}
+	keyPath, err := filepath.Abs(filepath.Join(certDir, defaultKubeletClientKeyFile))
+	if err != nil {
+		glog.Errorf("unable to build HTTPS prober key path: %v", err)
+		return NewHttpProber()
```

Comment:
Will it be confusing to return a HttpProber whenever an error occurs?

Reply:
Removed. In this case, we still create a HTTPSProber, and the probe will fail.